tide_site_restriction.module

```diff
@@ -371,6 +297,17 @@ function tide_site_restriction_form_alter(&$form, FormStateInterface $form_state
   }
 }
 
+/**
+ * Implements hook_user_login().
+ */
+function tide_site_restriction_user_login(UserInterface $account) {
+  $site_restriction_helper = \Drupal::service('tide_site_restriction.helper');
+  if (!$site_restriction_helper->getUserSites($account)) {
+    \Drupal::messenger()
```

Comment:
@vincent-gao `tide_site_preview` may be not available, hence always displaying this message after user login does not sound right. IMHO you should check for the preview module, or show the message on the block `tide_site_preview_links_block` https://github.com/dpc-sdp/tide_site/blob/develop/modules/tide_site_preview/src/Plugin/Block/PreviewLinksBlock.php